dashboard/public/js/main.js

```diff
@@ -260,6 +260,7 @@ function onLocalized() {
 		lang.onchange = function() {
 			l10n.setLanguage(this.value);
 			localStorage.setItem("languageSelection", this.value);
+			location.href = window.location.pathname + "?lang="+document.getElementById('languageSelection').value;
```

Comment:
@NikhilM98 

NO REFRESH!
This defeats the purpose of changing the language without refresh. Moreover, I didn't see this issue on dev.

Reply:
I compared the files of dev and master. They have same issue.

Dev:
![screenshot from 2019-03-07 17-47-11](https://user-images.githubusercontent.com/24666770/53956268-2e5b0300-4101-11e9-819f-998dc36f0dc4.png)

Master:
![screenshot from 2019-03-07 17-47-04](https://user-images.githubusercontent.com/24666770/53956283-361aa780-4101-11e9-80a2-9bb63455b7c0.png)

(Some of the text refrenced in this issue was not translated due to missing translations. The commit refrencing missing translations been already merged.)

The timestamp is not being translated because `moment.locale` defined in the server is used for translating timestamp. That's why I recommended refreshing the page. But now I understand that refreshing the page is a bad choice.

https://github.com/llaske/sugarizer-server/blob/a56dd9ac441df60faa9628778c69c07249eb5006/dashboard/controller/journal.js#L19-L22

I think getting timestamp language from the server is a also not a good idea. I can see a hack here. Since only one instance of moment is initialized in the project. A user can change the timestamp language for other users.

I'd suggest setting `moment.locale()` in front-end insted of back-end. I also recommend updating minified version. Current minified version is "2.5.1". There are major changes in moment.js after ["2.8.0"](https://github.com/moment/moment/blob/develop/CHANGELOG.md#280-see-changelog) which can be relevent to us.

What do you say @tarunsinghal92?

Reply:
Update: I'm trying to translate timestamp in front-end. You can see the progress [here](https://github.com/NikhilM98/sugarizer-server/commit/28bb44ab1d38d20a2c39cbb96e1b652b66a23238).
I'd to made more changes than expected. Please check the diff of last commit on `nikhilm98/moment-update` and review.

Reply:
I've not handled moment translations for the timestamps in the charts. The data for the charts is received from the backend, so I think we can modify the API for the charts. We can pass the language in the call, translate in the backend and receive translated data.